graphene_linked_events/schema.py

```diff
@@ -166,9 +170,21 @@ class Event(IdObject):
     provider_contact_info = String()
     description = Field(LocalisedObject)
     p_event = Field(PalvelutarjotinEventNode)
+    venue = Field(VenueNode)
 
     def resolve_p_event(self, info, **kwargs):
-        return PalvelutarjotinEvent.objects.get(linked_event_id=self.id)
+        try:
+            return PalvelutarjotinEvent.objects.get(linked_event_id=self.id)
+        except PalvelutarjotinEvent.DoesNotExist:
+            return None
+
+    def resolve_venue(self, info, **kwargs):
+        if self.location.id:
```

Comment:
When using a related object's id like this, I think it is generally better to use the actual foreign key field's value, like here `self.location_id`, so you will save db hit(s).

Reply:
Unfortunately there is no `location_id` in this `self` object because the Event obj is formatted from JSON REST response returned from LinkedEventAPI, we don't have the actual model of Event in Palvelutarjotin. `self.location.id` corresponds to  `event { location { id: ` in the json data from `LinkedEvent`

Reply:
Aaa right, then my comment is of course invalid here, sorry I missed that :)